### DISCUSSION TITLE
Firefox extenstion: Unable to detect 4K content

### DISCUSSION THREAD

**Discussion by JonstrupDK** [aclap-dev/video-downloadhelper#1304] [Help and Q&A]
### Your question

Hi, 

I'm running the latest version through Firefox. On Youtube, I am only able to detect and download content up to 1440p. How come I cannot download the 4k video version? 

Video is loading in 4k, so it is available. 

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"2bd4165f881e488d"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:130.0) Gecko/20100101 Firefox/130.0
{
    "valid_license_message_has_been_displayed": true,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 12,
    "successfull_dl": 1
}

**Comment by Mattmjs** [comment]
I see 4k on this one https://www.youtube.com/watch?v=FFzO5JbQO6c but only as MKV. Mp4 options don't have 4k.
Do you see 4k on this ?

**Reply by GrampaWildWilly** [reply]
Right.  Sometimes there's a 4K variant, sometimes there isn't.  It's entirely dependent on whoever uploaded the video.  When there is a 4K variant, VDH almost always can download it.  But like Matt said, you need to open the list of resolutions & scroll it to see whether there's an MKV variant at 4K resolution.

**Reply by JonstrupDK** [reply]
> I see 4k on this one https://www.youtube.com/watch?v=FFzO5JbQO6c but only as MKV. Mp4 options don't have 4k. Do you see 4k on this ?

Yes, I see 4k on that one as well. 

Strange, because a wide range of other videos I still don't. 

These are randomly selected from my recommended section. They stream in 4K but only 1440p available in Downloadhelper: 

https://www.youtube.com/watch?v=ltMHgHFee0k&t=281s&ab_channel=grainydays
https://www.youtube.com/watch?v=D7WI0pv8-FQ&t=7s&ab_channel=ShortCircuit

Could it be that they are merely uploaded as 1440p files, and then upscaled by Youtube to allow 4k option?

**Reply by GrampaWildWilly** [reply]
That is quite weird.  Both of those have a highest resolution of 720p for MP4 but they both have 1440p & 1080p variants as MKV.  In other words the set of MP4 variants does not show the same set of resolutions as the MKV variants.  It is typical for there to be a 2160p MKV but no 2016p MP4.  Ditto for 1440p.  However, it is unusual that there is no 1080p MP4 when there is a 1080p MKV.  Even playing the items in the web page at 4K does not make a 4K variant appear in VDH.  This is beyond my knowledge.  I suspect only @paulrouget Paul can explain it.  I do not have a 4K monitor so I have no way of verifying whether YouTube's claim that it is playing those videos at 4K is true.  It is possible that they're just saying it's 4K when it really isn't.  But I can't verify that.

There is one other possibility.  I quite often look for NHL game highlights on YouTube.  Sometimes, if I find the highlights soon enough after the game ends, the clip offers only a 360p variant.  After a few minutes, maybe half an hour, the full array of variants appears.  Now, your 2 videos appeared 16 & 19 hours ago, as I write this.  You would think that's plenty of time for all the variants to appear.  But maybe not.  Give it another day.  The missing 4K variants might appear.  But I wouldn't expect it to take more than 1 more day.  Actually, I would have expected the 4K variants to have appeared by now.  But it's YouTube.  I've never seen logical behavior on YouTube.

So, in many words, I've said . . . I dunno.
